### PR TITLE
Add sum / average to array to do minor refactor

### PR DESCRIPTION
**Pull request by jsbean**
Also moves `Benchmark` to own file, as it is more global than just necessary for `linearRegression`.

### CODE REVIEW

File: Sources/PerformanceTesting/PerformanceTestCase.swift [hunk 1]
```diff
@@ -26,7 +26,8 @@ open class PerformanceTestCase: XCTestCase {
 
     // MARK: Instance Methods
 
-    /// Benchmarks the performance of a closure.
+    /// - Returns: An array of two-tuples containing the size of the structure and the average time
```
Comment (mossheim) [inline Sources/PerformanceTesting/PerformanceTestCase.swift]: "size of the structure" seems too specific at this point - i feel like if we change it here then we should change the labels in the routine as well. can we keep it to `testPoint` for now?

Comment (jsbean) [reply Sources/PerformanceTesting/PerformanceTestCase.swift]: Yeah, I agree. I feel like there could be a middle ground between `testPoint` and `size`.

Comment (mossheim) [reply Sources/PerformanceTesting/PerformanceTestCase.swift]: per slack discussion, could you maybe just change the wording from `size of the structure` to `input size`? I'm planning a major refactor and the stuff you've added in this PR will be helpful.

Comment (jsbean) [reply Sources/PerformanceTesting/PerformanceTestCase.swift]: Fixed in d301941.

**Comment by mossheim** [comment]
ty!

**Pull request merged**